website/docs/stitch-directives-sdl.md

```diff
@@ -0,0 +1,385 @@
+  return async ({ document, variables }) => {
+    const query = typeof document === 'string' ? document : print(document);
+    const fetchResult = await fetch(url, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ query, variables }),
+    });
+    return fetchResult.json();
+  };
+}
+
+async function fetchRemoteSchema(executor) {
+  // 2. fetch schemas from their raw SDL queries...
+  const result = await executor({ document: '{ _sdl }' });
+  return buildSchema(result.data._sdl);
+}
+```
+1. Include the `stitchingDirectivesTransformer` in your stitched gateway's config transforms. This will read SDL directives into the stitched schema's static configuration.
+2. Fetch subschemas through their `_sdl` query. You _cannot_ introspect custom directives, so you must use a custom query that provides the complete annotated type definitions string.
+
+## Recipes
+
+### Picked keys
+
+The simplest merge pattern picks a key field from origin objects:
+
+```graphql
+type User @key(selectionSet: "{ id }") {
+  # ...
+}
+
+type Product @key(selectionSet: "{ upc }") {
+  # ...
+}
+
+type Query {
+  user(id: ID!): User @merge(keyField: "id")
+  products(upcs: [ID!]!): [Product]! @merge(keyField: "upc")
+}
+```
+
+This SDL translates into the following merge config:
+
+```js
+merge: {
+  User: {
+    // single-record query:
+    selectionSet: '{ id }'
+    fieldName: 'user',
+    args: ({ id }) => ({ id }),
+  },
+  Product: {
+    // array query:
+    selectionSet: '{ upc }'
+    fieldName: 'products',
+    key: ({ upc }) => upc,
+    argsFromKeys: (upcs) => ({ upcs }),
+  }
+}
+```
+
+Here the `@key` directive specifies a base selection set for each merged type, and the `@merge` directive marks each type's merge query&mdash;then `keyField` specifies a field to be picked from each original object as the query argument value.
+
+### Multiple arguments
+
+This pattern configures a merge query that receives multiple arguments:
+
+```graphql
+type User @key(selectionSet: "{ id }") {
+  # ...
+}
+
+type Query {
+  users(ids: [ID!]!, scope: String): [User]! @merge(
+    keyField: "id",
+    keyArg: "ids",
+    additionalArgs: """ scope: "all" """
+  )
+}
+```
+
+This SDL translates into the following merge config:
+
+```js
+merge: {
+  User: {
+    selectionSet: '{ id }'
+    fieldName: 'users',
+    key: ({ id }) => id,
+    argsFromKeys: (ids) => ({ ids, scope: 'all' }),
+  }
+}
+```
+
+Because the merge field recieves multiple arguments, the `keyArg` parameter is required to specify which argument recieves the key(s). The `additionalArgs` parameter may then be used to provide static values for the other arguments.
+
+### Object keys
+
+In the absence of a `keyField` to pick, keys will assume the shape of an object with a `__typename` and all fields collected for all selectionSets on the type. These object keys should be represented in your schema with a dedicated scalar type:
```

Comment:
maybe we can change should to could, and foreshadow the typed approach below?

Reply:
Revised to directly refer down to the typed input section below.